### PR TITLE
Issue #10548: Migrate to Inline Config Parser in ArrayTrailingCommaCheckTest

### LINKED ISSUE #10548: Migrate to Inline Config Parser in Coding Check Tests
Child of #10390

item:
- [x] ArrayTrailingCommaCheckTest - #10549
- [x] AvoidDoubleBraceInitializationCheckTest - #10568
- [x] AvoidInlineConditionalsCheckTest - #10569
- [x] AvoidNoArgumentSuperConstructorCallCheckTest - #10570
- [x] CovariantEqualsCheckTest - #10571
- [x] DeclarationOrderCheckTest - #10572
- [x] DefaultComesLastCheckTest - #10573
- [x] EmptyStatementCheckTest - #10578
- [x] EqualsAvoidNullCheckTest - #10581
- [x] EqualsHashCodeCheckTest - #10606
- [x] ExplicitInitializationCheckTest - #10639
- [x] FallThroughCheckTest - #10643
- [x] FinalLocalVariableCheckTest - #10643
- [x] HiddenFieldCheckTest - #10644
- [x] IllegalCatchCheckTest - #10610
- [x] IllegalInstantiationCheckTest - #10611
- [x] IllegalThrowsCheckTest - #10612
- [x] IllegalTokenCheckTest - #10613
- [x] IllegalTokenTextCheckTest - #10615
- [x] IllegalTypeCheckTest - #10614 
- [x] InnerAssignmentCheckTest - #10616
- [x] MagicNumberCheckTest - #10645
- [x] MatchXpathCheckTest - #10648
- [x] MissingCtorCheckTest - #10621
- [x] MissingSwitchDefaultCheckTest - #10622
- [x] ModifiedControlVariableCheckTest - #10630
- [x] MultipleStringLiteralsCheckTest - #10632
- [x] MultipleVariableDeclarationsCheckTest - #10646
- [x] NestedForDepthCheckTest - #10633
- [x] NestedIfDepthCheckTest - #10634
- [x] NestedTryDepthCheckTest - #10635
- [x] NoArrayTrailingCommaCheckTest - #10641
- [x] NoCloneCheckTest - #10641
- [x] NoEnumTrailingCommaCheckTest - #10641
- [x] NoFinalizerCheckTest - #10641
- [x] OneStatementPerLineCheckTest - #10641
- [x] OverloadMethodsDeclarationOrderCheckTest - #10641
- [x] PackageDeclarationCheckTest - #10646
- [x] ParameterAssignmentCheckTest - #10641
- [x] RequireThisCheckTest - #10646
- [x] ReturnCountCheckTest - #10647
- [x] SimplifyBooleanExpressionCheckTest - #10641
- [x] SimplifyBooleanReturnCheckTest - #10641
- [x] StringLiteralEqualityCheckTest - #10641
- [x] SuperCloneCheckTest - #10641
- [x] SuperFinalizeCheckTest - #10641
- [x] UnnecessaryParenthesesCheckTest - #10647
- [x] UnnecessarySemicolonAfterOuterTypeDeclarationCheckTest - #10641
- [x] UnnecessarySemicolonAfterTypeMemberDeclarationCheckTest - #10641
- [x] UnnecessarySemicolonInEnumerationCheckTest - #10641
- [x] UnnecessarySemicolonInTryWithResourcesCheckTest - #10641
- [x] VariableDeclarationUsageDistanceCheckTest - #10641

### PR DESCRIPTION
**Pull request by shashwatj07**
part of #10548

### CODE REVIEW

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/arraytrailingcomma/InputArrayTrailingComma.java [hunk 1]
```diff
@@ -20,7 +20,7 @@
     {
         1,
         2,
-        3 // violation
+        3 // violation Array should contain trailing comma.
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/arraytrailingcomma/InputArrayTrailingComma.java:23]: dream is close to completion:
make it like `// violation 'Array should contain trailing comma.'` to let us has some structure in comments

Comment (shashwatj07) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/arraytrailingcomma/InputArrayTrailingComma.java:23]: This will require modification of our regexp.

Let's also discuss what to do if we have multiple violations on same line.
Proposal:
```// violation 'violation message 1', 'violation message 2', 'violation message 3'```
Just one con is that it'll result in very long lines in some cases.

Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/arraytrailingcomma/InputArrayTrailingComma.java:23]: We will do our best to make all lines to have single violation.

If not possible, such edge cases will use old style verification of expected violations in addition to trailing comments validation

Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/arraytrailingcomma/InputArrayTrailingComma.java:23]: Keep going with updates on simple cases and move all non simple to separate issues, to deal with them later on . So at least we will see all cases and it will be easier to make decisions on design

Comment (shashwatj07) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/arraytrailingcomma/InputArrayTrailingComma.java]: > make it like // violation 'Array should contain trailing comma.' to let us has some structure in comments

done

**Comment by romani** [comment]
Tc is not happy, please fix

**Review by romani — APPROVED** [review]
